lib/std/Build/Step/RemoveDir.zig

```diff
@@ -20,6 +20,7 @@ pub fn create(owner: *std.Build, doomed_path: LazyPath) *RemoveDir {
         }),
         .doomed_path = doomed_path.dupe(owner),
     };
+    doomed_path.addStepDependencies(&remove_dir.step);
```

Comment:
FYI, since Andrew has veoted this change we might want to add this bug fix with another PR.  There are more big fixes like this in the planned follow up PR but without my changes they won't be detected, we have to manually find them with my branch and then port them over to upstream.

Reply:
I think I found one instance where there was more missing lazy path dependencies in `CheckObject.zig`:
https://github.com/ziglang/zig/compare/master...marler8997:zig:lazyPathMisuse#diff-250b661d4bed9463d5fef7e603cd49501d1ffb8b439330ce818984247abb1552